### DISCUSSION TITLE
Version 0.4.1 never searches web

### DISCUSSION THREAD

**Discussion by Expro** [open-webui/open-webui#7096] [Issues]
## Environment

Open WebUI 0.4.1 in container
Ollama 0.4.2

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [-] I have included the browser console logs.
- [-] I have included the Docker container logs.
- [X] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

When web search is enabled and configured properly (same setup worked correctly on 0.3.5), Open WebUI should generate web search queries and search websites.

## Actual Behavior:

UI says "No search query generated" and no web search is performed.

**Steps to Reproduce:**

1. Setup web search on 0.3.5 and verify that it works correctly.
2. Upgrade do 0.4.1
3. Perform same actions that previously initiated web search.

**Comment by comeback01** [comment]
I have exactly the same problem. Before the update, everything was working fine. Now it doesn't work. I tested with SearxNG and DuckDuckGo, which worked perfectly before.

**Comment by tjbck** [comment]
![IMG_2007](https://github.com/user-attachments/assets/3ec95b13-7ea6-47b3-a26b-e5d7aa0424c6)

Unable to reproduce the issue here, keep us posted!

**Comment by comeback01** [comment]
I looked at your screenshot and noticed that you’re using GPT-4o. I tested it with DuckDuckGo, and it works as well.

However, with https://groq.com/ and Gemini via OpenRouter, it doesn’t work.

With SearxNG, GPT-4o also doesn’t work.

For your information, everything was working fine before.

**Reply by Mushy-Snugglebites-badonkadonk** [reply]
DDG and SearXNG isn’t working in 0.4.5. The search will generate and return links, RAG is then broken as no information is read and therefore no citations. 

What is interesting is when you are searching for a second term the original search term is searched again.

I don’t think that temporary mode should affect this issue, but just putting that out there as well.
![image](https://github.com/user-attachments/assets/4ec7e41f-04a5-4a99-99e6-e82e7af490ba)

**Comment by Expro** [comment]
My setup (that is NOT working) is Ollama + SearXNG.

**Reply by tjbck** [reply]
Definitely works with Ollama on my end:

![image](https://github.com/user-attachments/assets/5c8c81ad-a53a-4a68-92b0-5b74f75c1e63)

**Reply by Mushy-Snugglebites-badonkadonk** [reply]
Web Search & Personal Memory both are not working. I am running Ollama as a separate docker container. Screenshot shows my Documents settings. I’ve cleared Upload and Vector data. I zeroed out _minimum score_ and the issue still persists. The Open WebUI will search the internet via a local SearXNG instance, and will show that Open AI searched websites, but Open WebUI’s response is from the model’s dataset; RAG is broken. 

Version: 0.4.2

Errors
`HTTPConnectionPool(host='host.docker.internal', port=11434): Max retries exceeded with url: /api/embed (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0x1473bef95a10>: Failed to resolve 'host.docker.internal' ([Errno -2] Name or service not known)")) `

` File "/app/backend/open_webui/apps/retrieval/utils.py", line 307, in <lambda>
    return lambda query: generate_multiple(query, func)
HTTPConnectionPool(host='host.docker.internal', port=11434): Max retries exceeded with url: /api/embed (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0x1473c21ba690>: Failed to resolve 'host.docker.internal' ([Errno -2] Name or service not known)"))                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^`

 ` File "/app/backend/open_webui/apps/retrieval/utils.py", line 305, in generate_multiple
    return func(query)
           ^^^^^^^^^^^`

`{'name': 'bge-m3:567m', 'model': 'bge-m3:567m', 'modified_at': '2024-11-01T09:14:48.454623209Z', 'size': 1157672605, 'digest': '7907646426070047a77226ac3e684fbbe8410524f7b4a74d02837e43f2146bab', 'details': {'parent_model': '', 'format': 'gguf', 'family': 'bert', 'families': ['bert'], 'parameter_size': '566.70M', 'quantization_level': 'F16'}}]}] `

![IMG_0432](https://github.com/user-attachments/assets/d280fecd-3731-4185-b5a6-13d2611a644e)

**Reply by tjbck** [reply]
@Mushy-Snugglebites-badonkadonk Your issue seems to be caused by docker network misconfiguration and your container not being able to reach `host.docker.internal`. Also, I'd opt-in to use the default RAG template by removing your custom prompt and see if that resolves the issue. Keep us updated!

**Reply by Mushy-Snugglebites-badonkadonk** [reply]
The issue has been resolved. I did not catch the server IP address change with the UI update. Inputing the IP address of the Unraid server and corresponding port resolved the issue. Apologies for contributing noise.

Removing the old default RAG template for the new default is working well.

**Comment by tjbck** [comment]
I'm not 100% certain but `SearxNG` might be the culprit here. It's strange because we have not changed anything in regards to web search since the 0.3 release. Keep us updated on your troubleshooting journey!

**Reply by Mushy-Snugglebites-badonkadonk** [reply]
Everything was working prior to the 0.4.x updates released on 19-Nov-24

**Reply by comeback01** [reply]
Did you read my message?

I looked at your screenshot and noticed that you’re using GPT-4o. I tested it with DuckDuckGo, and it works as well.

However, with https://groq.com/ and Gemini via OpenRouter, it doesn’t work.

With SearxNG, GPT-4o also doesn’t work.

For your information, everything was working fine before.

**Comment by Expro** [comment]
![image](https://github.com/user-attachments/assets/2bfb6d84-38b1-4ca8-b47b-12bb30af5620)

That's on my setup (Ollama + SearXNG).

**Reply by tjbck** [reply]
<img width="1266" alt="image" src="https://github.com/user-attachments/assets/3524f01c-448b-49e9-baa6-b183dd0a881a">

1.5b models most likely won't be able to generate search queries, try disabling this option.

**Reply by 3c2b2ff5** [reply]
I have a 7B model running and I had to disable it as well

**Comment by Expro** [comment]
No errors in Open WebUI or SearXNG logs.

**Comment by hillzzz** [comment]
I have the same issue, No web tools work, in 0.4.1 Running Ollama + DuckDuck-go search. I have tried all the network settings and permissions I can think of but nothing works.

**Comment by comeback01** [comment]
On a new installation, I have no issues at all!

Since I am running a container from the Docker image, there shouldn't be any problem on that side.

However, I notice that the .env file no longer affects these parameters.

The only difference between the two configurations seems to be the database.

I think the issue comes from there, but it's just a gut feeling—nothing certain!

My problem is that I have a server with multiple users, and I can't start from scratch.

**Reply by Mushy-Snugglebites-badonkadonk** [reply]
Hmmm… I don’t like the sound of that. Sounds like a glitch. We shouldn’t have to reset the database to zero. 

Something in OWUI stopped working as LLMs are not able to read any of the Personalization Memories in addition to RAG not working with web search.

**Reply by comeback01** [reply]
The program only takes into account what is in the interface; it does not consider these parameters:


ENABLE_RAG_WEB_SEARCH=True
ENABLE_SEARCH_QUERY=True
RAG_WEB_SEARCH_ENGINE=duckduckgo
RAG_WEB_SEARCH_RESULT_COUNT=5
RAG_WEB_SEARCH_CONCURRENT_REQUESTS=5

**Reply by wengzhiwen** [reply]
YES METOO
After 5hours' check-review-google-fix-check.
docker rm open-webui
fixed everything, just like a Microsoft Windows.

Azure OpenAI API (GPT-4o) + local Searxng

**Reply by tjbck** [reply]
Env vars are only used when the instance is first created. Afterwards, the config data is retrieved from the database.

**Comment by comeback01** [comment]
![Capture d'écran 2024-11-23 125334](https://github.com/user-attachments/assets/bd5b8f25-b781-4640-ba27-5174371d4c7c)
I conducted several tests, and here is some new information:

1) When I perform the search in English, it works.  
2) When I perform the search in French, it doesn’t work.  

See the image.

**Reply by wengzhiwen** [reply]
It's not stable. Sometime works sometime not (the same query the same language)
Tested in Chinese and Japanese, query for tomorrow's weather.

**Comment by comeback01** [comment]
⚠️⚠️ Solution ⚠️⚠️

By replacing the embedding model with a multilingual one, it works.

I used this model: "embaas/sentence-transformers-multilingual-e5-base". I'm not sure if it's the best choice, but it provides good results.

The default model is designed only for English.

**Reply by Expro** [reply]
It's not a solution for this issue. I created this bug report while only using English queries. If anything, this suggests there are **several** bugs at play.

**Comment by tjbck** [comment]
Related: https://github.com/open-webui/open-webui/discussions/7105#discussioncomment-11386749

**Comment by mikygit** [comment]
any news on this 'bug'?
